lib/auth/webauthncli/api.go

```diff
@@ -23,10 +23,23 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+// LoginOpts groups non-mandatory options for Login.
+type LoginOpts struct {
+	// User is the desired credential username for login.
+	// If empty, Login may either choose a credential or error due to ambiguity.
+	User string
+	// OptimisticAssertion allows Login to skip credential listing and attempt
+	// to assert directly. The drawback of optimistic assertions is that Login is
+	// unable to guarantee the credential user, even if the User field is set.
```

Comment:
I can see how that would sound strange.

When we do an optimistic assertion we let the authenticator pick the credential. If the authenticator has multiple credentials for that cluster, we don't know which one it will pick, so we can't guarantee that the credential is for the user in LoginOpts.User.

Does that answer your question? If yes I'll tweak the docs to better explain it.